rust/parquet/src/arrow/schema.rs

```diff
@@ -657,6 +645,22 @@ impl ParquetTypeConverter<'_> {
         }
     }
 
+    fn to_decimal(&self) -> Result<DataType> {
+        let (precision, scale) = match self.schema {
+            Type::PrimitiveType {
+                ref precision,
+                ref scale,
+                ..
+            } => (*precision, *scale),
+            _ => {
+                return Err(ArrowError(
```

Comment:
I think you can use assertion here since this is only supposed to be called on primitive types. And it's more concise to do something like this:

```rust
        match self.schema {
            Type::PrimitiveType {
                ref precision,
                ref scale,
                ..
            } => Ok(DataType::Decimal(*precision as usize, *scale as usize)),
            _ => Err(ArrowError(
                "Expected a physical type, not a group type".to_string(),
            )),
        }
```

Reply:
@sunchao thank you for your review comments! I changed this part as suggested. Do you have an example of an assertion checking for a type? The only thing I could come up with (still quite new to Rust) is something along the lines of

```
if let Type::PrimitiveType {ref precision, ref scale, .. } = self.schema {
  DataType::Decimal(*precision as usize, *scale as usize)
} else {
  panic!("Expected a physical type, not a group type")
}
```

Reply:
I'm thinking something like:
```rust
assert!(self.schema.is_primitive());
if let ... {
  ...
}
```

Reply:
Thank you very much! I changed it to use an assertion. Let me know if you prefer this variant - there are other methods in `schema.rs` that follow a similar pattern and I could also change them accordingly while I am already here.

Reply:
Thanks. Looking good! the `else` branch in the `if let` block seems unnecessary?